osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderSelectionBlueprint.cs

```diff
@@ -277,9 +281,9 @@ private void adjustLength(double proposedDistance, bool adjustVelocity)
             if (Precision.AlmostEquals(proposedDistance, HitObject.Path.Distance) && Precision.AlmostEquals(proposedVelocity, HitObject.SliderVelocityMultiplier))
                 return;
 
-            HitObject.SliderVelocityMultiplier = proposedVelocity;
-            HitObject.Path.ExpectedDistance.Value = proposedDistance;
-            editorBeatmap?.Update(HitObject);
+            Proxy.SetSliderVelocityMultiplier(proposedVelocity);
+            Proxy.Path().SetExpectedDistance(proposedDistance);
+            editorBeatmap?.AsCommandProxy(commandHandler).Update(HitObject);
```

Comment:
`editorBeatmap?.Update(HitObject);` adds the hitobject to a list to be batch-processed at a later time, the order in this case shouldn't matter as when undoing/redoing this will happen after the transaction has been fully applied.